samples/dnn/object_tracker.cpp

```diff
@@ -0,0 +1,330 @@
+/*
+For usage download models by following links
+    For DaSiamRPN:
+        network:     https://www.dropbox.com/s/rr1lk9355vzolqv/dasiamrpn_model.onnx?dl=0
+        kernel_r1:   https://www.dropbox.com/s/999cqx5zrfi7w4p/dasiamrpn_kernel_r1.onnx?dl=0
+        kernel_cls1: https://www.dropbox.com/s/qvmtszx5h339a0w/dasiamrpn_kernel_cls1.onnx?dl=0
+    For NanoTrack:
+        nanotrack_backbone: https://github.com/HonglinChu/SiamTrackers/blob/master/NanoTrack/models/nanotrackv2/nanotrack_backbone_sim.onnx
+        nanotrack_headneck: https://github.com/HonglinChu/SiamTrackers/blob/master/NanoTrack/models/nanotrackv2/nanotrack_head_sim.onnx
+    For VitTrack:
+        vitTracker: https://github.com/opencv/opencv_zoo/raw/fef72f8fa7c52eaf116d3df358d24e6e959ada0e/models/object_tracking_vittrack/object_tracking_vittrack_2023sep.onnx
+*/
+
+#include <iostream>
+#include <cmath>
+
+#include <opencv2/dnn.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/highgui.hpp>
+#include <opencv2/video.hpp>
+#include "common.hpp"
+
+using namespace cv;
+using namespace std;
+using namespace cv::dnn;
+
+const string about = "Use this script for Object Tracking using OpenCV. \n\n"
+        "To run:\n"
+            "\t Nano: \n"
+                "\t\t e.g: ./example_dnn_object_tracker nano --backbone=<path to nanotrack_backbone onnx model> --headneck=<path to nanotrack_head onnx model>\n\n"
```

Comment:
The path of the models are confusing. Warning messages require setting `OPENCV_SAMPLES_DATA_PATH` or `OPENCV_DOWNLOAD_CACHE_DIR`; when `OPENCV_SAMPLES_DATA_PATH` is set, another warning appear: FileNotFoundError: File xxxx not found! Please specify the path to /opencv/samples/data in the OPENCV_SAMPLES_DATA_PATH environment variable, or specify the path to opencv_extra/testdata in the OPENCV_DNN_TEST_DATA_PATH environment variable, or specify the path to the model download cache directory in the OPENCV_DOWNLOAD_CACHE_DIR environment variable, or pass the full path to xxxx.

Reply:
@kaingwade The error message comes from findFile() function which searches for files in these environment variables. I have removed the default arguments, and added check and custom messages in case the model files are not specified